api/api/versions_api.go

```diff
@@ -100,17 +99,22 @@ func (c *VersionsController) ListVersions(r *http.Request, vars map[string]strin
 	return OkWithHeaders(versions, responseHeaders)
 }
 
-func (c *VersionsController) CreateVersion(r *http.Request, vars map[string]string, _ interface{}) *Response {
+func (c *VersionsController) CreateVersion(r *http.Request, vars map[string]string, body interface{}) *Response {
 	ctx := r.Context()
 
+	versionPost, ok := body.(*models.VersionPost)
+	if !ok {
+		return InternalServerError("Unable to parse request body")
```

Comment:
Do you think BadRequest is a more appropriate HTTP status here since it is error due to failure in parsing request payload?

Reply:
agreed. this should be a 4xx rather than 5xx error